tests/integration-tests/tests/tags/test_tag_propagation.py

```diff
@@ -0,0 +1,154 @@
+# Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License").
+# You may not use this file except in compliance with the License.
+# A copy of the License is located at
+#
+# http://aws.amazon.com/apache2.0/
+#
+# or in the "LICENSE.txt" file accompanying this file.
+# This file is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, express or implied.
+# See the License for the specific language governing permissions and limitations under the License.
+# import logging
+
+import json
+import logging
+
+import boto3
+import pytest
+from assertpy import assert_that
+
+from pcluster.utils import get_installed_version
+
+
+@pytest.mark.regions(["us-east-1"])
+@pytest.mark.instances(["c5.xlarge"])
+@pytest.mark.oss(["alinux2"])
+@pytest.mark.schedulers(["slurm", "torque", "awsbatch"])
+@pytest.mark.usefixtures("region", "os", "instance")
+def test_tag_propagation(pcluster_config_reader, clusters_factory, scheduler):
+    """
+    Verify tags specified via the config file and the command line are propagated to the expected resources.
+
+    Specifically, ensure the following resources are tagged:
+    - head node
+    - compute nodes
+    - EBS volumes
+    """
+    config_file_tags = {"ConfigFileTag": "ConfigFileTagValue"}
+    command_line_tags = {"CommandLineTag": "CommandLineTagValue"}
+    default_tags = {"Version": get_installed_version()}
+    cluster_config = pcluster_config_reader(tags=json.dumps(config_file_tags))
+    cluster = clusters_factory(cluster_config, extra_args=["--tags", json.dumps(command_line_tags)])
+
+    expected_tags = []
+    for tag_set in (config_file_tags, command_line_tags, default_tags):
+        expected_tags.extend([{"Key": tag, "Value": value} for tag, value in tag_set.items()])
+
+    test_cases = [
+        {"resource": "Main CloudFormation Stack", "tag_getter": get_main_stack_tags},
+        {"resource": "Head Node CloudFormation Stack", "tag_getter": get_head_node_substack_tags},
+        {
+            "resource": "ComputeFleet CloudFormation Stack",
+            "tag_getter": get_compute_fleet_substack_tags,
+            "tag_getter_args": (cluster, scheduler),
+        },
+        {"resource": "Head Node", "tag_getter": get_head_node_tags},
+        {"resource": "Compute Node", "tag_getter": get_compute_node_tags, "skip": scheduler == "awsbatch"},
+        {"resource": "EBS Volume", "tag_getter": get_shared_volume_tags},
+    ]
+    for test_case in test_cases:
+        if test_case.get("skip"):
+            continue
+        logging.info("Verifying tags were propagated to %s", test_case.get("resource"))
+        tag_getter = test_case.get("tag_getter")
+        # Assume tag getters use lone cluster object arg if none explicitly given
+        tag_getter_args = test_case.get("tag_getter_args", (cluster,))
```

Comment:
minor: wondering if we can use named args just so it's harder to mess up with the order

Reply:
i turned them into kwargs. i'm not sure if that's what you were looking for here though?